Intel_SighPP_CMake/Intel_SighPP_CMake/src/priority_lib/smart_priority.hpp

```diff
@@ -1,76 +1,155 @@
 #pragma once
 
 #include "priority_module.hpp"
+#include "priority_clock.hpp"
 #include <spdlog/spdlog.h>
+#include "../classification_result.hpp"
 #include <string>
+#include <map>
 
 class smart_priority : public priority_module
 {
 protected:
 	int const size_w = 640;
 	int const size_h = 480;
+	int const cooldown = 5000;
+	int const cooldown_shortening = 1000;
+	int const cooldown_emergency = 2500;
+	double const walking_speed = 1.4;
 	int centre_line = 0;
 	int out_left = 0;
 	int out_right = 0;
 	int in_left = 0;
 	int in_right = 0;
 	int above = 0;
-	int bottom  = 0;
+	int bottom = 0;
 	int mid_w = 0;
+	priority_clock pc;
+	Priority min = Priority::UNDEFINED;
 
 public:
 
+	/// <summary>
+	///Object Creation. Not to be inherited furhter.
+	/// Initialises location markers based on ratio
+	/// </summary>
+	/// <param name="name"></param>
+	/// <returns></returns>
 	smart_priority(std::string name = "default")
 		: priority_module(&name) {
 		determine_location_markers();
 	}
-
+	/// <summary>
+	/// Determines location markers for objects within the screen. 
+	/// </summary>
 	void determine_location_markers();
+
+
+	//________data and message control________: 
+
+	/// <summary>
+	/// Based on rectangle of object determin location
```

Comment:
Is this good english? 😄 
Also, if the item param is not documented, maybe include something about it in the summary 😀

Reply:
Fixed.